src/com/company/models/Artist.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * To change this template, choose Tools | Templates
+ * and open the template in the editor.
+ */
+
+package com.company.models;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toList;
+
+/**
+ * Domain class for a popular music artist.
+ * 
+ * @author Richard Warburton
+ */
+public final class Artist {
+    
+    private String name;
+    private List<Artist> members;
+    private String nationality;
+    
+    public Artist(String name, String nationality) {
+        this(name, Collections.emptyList(), nationality);
+    }
+
+    public Artist(String name, List<Artist> members, String nationality) {
+        Objects.requireNonNull(name);
+        Objects.requireNonNull(members);
+        Objects.requireNonNull(nationality);
+        this.name = name;
+        this.members = new ArrayList<>(members);
+        this.nationality = nationality;
+    }
+
+    /**
+     * @return the name
```

Comment:
JavaDoc for accessors (getters and setters)  is not necessary.

Reply:
Done